### PR TITLE
Fix empty tendermint version in Docker

### LINKED ISSUE #8160: Empty tendermint version in Docker
**Tendermint version**: [master](https://github.com/tendermint/tendermint/tree/485c96b0d386d5461d397fef76ae75df0d56c885), [v0.35.2](https://github.com/tendermint/tendermint/tree/v0.35.2) - docker images

**What happened**: an empty string is returned when you call `tendermint version` for tendermint in docker

**What you expected to happen**: the version of tendermint from the docker container should be printed

**Have you tried the latest version**: yes

**How to reproduce it**:
1. Build docker image from local repository or pull it from docker hub:

```sh
make build-docker # or: docker pull tendermint/tendermint:latest
```

2. Run tendermint docker container with `version` subcommand:

```sh
docker run --rm tendermint/tendermint version
```

3. Make sure that instead of version an empty string was returned

### PR DESCRIPTION
**Pull request by svkirillov**
Closes #8160

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -243,10 +243,8 @@ build-docs:
 ###                            Docker image                                 ###
 ###############################################################################
 
-build-docker: build-linux
-	cp $(BUILDDIR)/tendermint DOCKER/tendermint
+build-docker:
 	docker build --label=tendermint --tag="tendermint/tendermint" -f DOCKER/Dockerfile .
-	rm -rf DOCKER/tendermint
```
Comment (williambanfield) [inline Makefile:249]: Would you mind explaining why this change is ok?

**Comment by williambanfield** [comment]
Thanks for the fix.

It looks like the variable you actually should be targeting is the `TMVersion` var instead of the `TMVersionDefault` var.
https://github.com/tendermint/tendermint/blob/485c96b0d386d5461d397fef76ae75df0d56c885/version/version.go#L6
The `TMVersion` is what's ultimately used, and I'm not sure I see a reason to overwrite the `TMVersionDefault` variable. 

Looks like this is what the old `TMCoreSemVer` variable was as well, so must as well keep this consistent.

File: Makefile (hunk 1 above)
Comment (svkirillov) [reply Makefile:249]: The [DOCKER/Dockerfile](https://github.com/tendermint/tendermint/blob/485c96b0d386d5461d397fef76ae75df0d56c885/DOCKER/Dockerfile) is multi-stage and [the fist stage](https://github.com/tendermint/tendermint/blob/485c96b0d386d5461d397fef76ae75df0d56c885/DOCKER/Dockerfile#L1-L8) is building a tendermint binary in docker and [the second](https://github.com/tendermint/tendermint/blob/485c96b0d386d5461d397fef76ae75df0d56c885/DOCKER/Dockerfile#L42) is building resulting docker image with the binary from the previous stage. Because of this, I think that another linux binary build process is redundant, and furthermore built linux binary is not used in the docker image building process.

**Comment by svkirillov** [comment]
> Thanks for the fix.
> 
> It looks like the variable you actually should be targeting is the `TMVersion` var instead of the `TMVersionDefault` var.
> 
> https://github.com/tendermint/tendermint/blob/485c96b0d386d5461d397fef76ae75df0d56c885/version/version.go#L6
> 
> The `TMVersion` is what's ultimately used, and I'm not sure I see a reason to overwrite the `TMVersionDefault` variable.
> 
> Looks like this is what the old `TMCoreSemVer` variable was as well, so must as well keep this consistent.

If I understand correctly, `TMVersion` overwritten with `git describe` when we build binary file with make to get the most accurate version. If we build tendermint directly with `go build` we get version in TMVersion from `TMVersionDefault`. I don't know if anyone is currently using the DOCKER/build.sh and DOCKER/push.sh scripts, but it contained an errors ([1](https://github.com/tendermint/tendermint/pull/8161/files#diff-6df363a1166519980e7a381be2b94cb338461388d289bd1923627763b9be4be5L6), [2](https://github.com/tendermint/tendermint/pull/8161/files#diff-b1ad61c089be06446a47e459d9e7574699588cc9fd59c495216a1a92389dbed7L6)). Perhaps awk should be replaced with git describe.

**Comment by williambanfield** [comment]
>  I don't know if anyone is currently using the DOCKER/build.sh and DOCKER/push.sh scripts

The only users in the repository are the targets within the [./DOCKER/Makefile](https://github.com/tendermint/tendermint/blob/e2365c1368c25bfbbd1404c49a9855a9677bb222/DOCKER/Makefile#L5). Looking through the git history, these were added before we switched to using github workflows for pushing to a docker repository. At this point, it looks like the correct thing to do is to remove the `DOCKER/push.sh` and `DOCKER/build.sh` files.

**Comment by svkirillov** [comment]
> At this point, it looks like the correct thing to do is to remove the `DOCKER/push.sh` and `DOCKER/build.sh` files.

Looks like [DOCKER/Dockerfile.build_c-amazonlinux](https://github.com/tendermint/tendermint/blob/master/DOCKER/Dockerfile.build_c-amazonlinux) and [DOCKER/Dockerfile.testing](https://github.com/tendermint/tendermint/blob/master/DOCKER/Dockerfile.testing) are not needed either. If it's true, then I'll delete it too.

File: Makefile (hunk 1 above)
Comment (williambanfield) [reply Makefile:249]: Thanks for the explanation, that definitely makes sense. Thanks for catching that.

File: DOCKER/Dockerfile [hunk 2]
```diff
@@ -2,7 +2,7 @@
 FROM golang:1.17-alpine as builder
 RUN apk update && \
     apk upgrade && \
-    apk --no-cache add make
+    apk --no-cache add make git
```
Comment (williambanfield) [inline DOCKER/Dockerfile:5]: Why is git needed here?

**Review by williambanfield — CHANGES_REQUESTED** [review]
Thanks for cleaning these up. The removed `Dockerfiles` appear to have been historically used as part of test infrastructure that we no longer use.
The Amazon linux container was used as part of load testing infrastructure. https://github.com/tendermint/tendermint/pull/3976

The `Dockerfile.testing` appears to have been used as part of bash test script: https://github.com/tendermint/tendermint/pull/1529

Other than the `git` addition in `DOCKER/Dockerfile` (which may be correct, I'm not yet sure), this looks great.

File: DOCKER/Dockerfile (hunk 2 above)
Comment (svkirillov) [reply DOCKER/Dockerfile:5]: Git is needed here because when we build binary with make it uses git to get the actual version. In the beginning of the [Makefile](https://github.com/tendermint/tendermint/blob/41a1bf539bd9e3f312e5a52d2424de41d646be21/Makefile#L7-L14) we have the [LD_FLAGS](https://github.com/tendermint/tendermint/blob/41a1bf539bd9e3f312e5a52d2424de41d646be21/Makefile#L14) variable setting which allows us to change the variable in the source code to set the actual tendermint version based on the git information. If git was not found during make running, the value of $(VERSION) will be empty, and because of this `tendermint version` will return an empty string instead of correct value. This is the reason why we need to install git in docker before building the tendermint binary.

**Review by williambanfield — APPROVED** [review]
